mutiny-core/src/lib.rs

```diff
@@ -355,6 +532,201 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         });
     }
 
+    /// Pays a lightning invoice from a federation (preferred) or node.
+    /// An amount should only be provided if the invoice does not have an amount.
+    /// Amountless invoices cannot be paid by a federation.
+    /// The amount should be in satoshis.
+    pub async fn pay_invoice(
+        &self,
+        inv: &Bolt11Invoice,
+        amt_sats: Option<u64>,
+        labels: Vec<String>,
+    ) -> Result<MutinyInvoice, MutinyError> {
+        if inv.network() != self.config.network {
+            return Err(MutinyError::IncorrectNetwork(inv.network()));
+        }
+
+        // Try each federation first
+        let federation_ids = self.list_federations().await?;
+        for federation_id in federation_ids {
+            if let Some(fedimint_client) = self.federations.lock().await.get(&federation_id) {
+                // Check the amount specified in the invoice
+                if let Some(invoice_amount) = inv.amount_milli_satoshis() {
```

Comment:
should do this check before `list_federations` otherwise we'll have to loop through each time for no reason